pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/HLRealtimeSegmentDataManager.java

```diff
@@ -123,7 +123,9 @@ public HLRealtimeSegmentDataManager(final RealtimeSegmentZKMetadata segmentMetad
     }
     //inverted index columns
     invertedIndexColumns = indexingConfig.getInvertedIndexColumns();
-
+    if (sortedColumn != null && !invertedIndexColumns.contains(sortedColumn)) {
+      invertedIndexColumns.add(sortedColumn);
```

Comment:
We shouldn't need this, right? The server does not load the inverted index if a column is sorted, afaik